solutions/gold/cf-510C.mdx

```diff
@@ -0,0 +1,84 @@
+---
+id: cf-510C
+source: CF
+title: Fox And Names
+author: Daniel Ge
+---
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// used for topological sort
+int indegree[26];
+vector<int> conn[26];
+vector<int> sorted;
+
+void topological_sort() {
+	queue<int> q;
+	for (int i = 0; i < 26; i++) {
+		if (indegree[i] == 0) q.push(i);
+	}
+	while (!q.empty()) {
+		int pos = q.front();
+		q.pop();
+		sorted.push_back(pos);
+		for (auto c : conn[pos]) {
+			if ((--indegree[c]) == 0) q.push(c);
+		}
+	}
+}
+
+int main() {
+	ios_base::sync_with_stdio(0);
+	cin.tie(nullptr);
+	int N;
+	string cur, prev;  // we make sure each pair of consecutive strings are in
```

Comment:
put the comments before and use multiline. also make sure that the lines are evenly spaced.